File: frontend/autocare-frontend/src/services/bookingService.js
import api from "./api";

const bookingService = {

    // Get all bookings
    getAllBookings() {
        return api.get("/bookings");
    },

    // Get booking by ID
    getBookingById(id) {
        return api.get(`/bookings/${id}`);
    },

    // Create booking
    createBooking(booking) {
        return api.post("/bookings", booking);
    },

    // Update booking status
    updateBookingStatus(id, status) {
        return api.put(
            `/bookings/${id}/status?status=${status}`
        );
    },

    // Delete booking
    deleteBooking(id) {
        return api.delete(`/bookings/${id}`);
    },

    // Get bookings by email
    getBookingsByEmail(email) {
        return api.get(`/bookings/email/${email}`);
    },

    // Get bookings by status
    getBookingsByStatus(status) {
        return api.get(`/bookings/status/${status}`);
    }

};

export default bookingService;

File: autocare-backend/src/main/java/com/pobitra/autocare/service/BookingService.java
package com.pobitra.autocare.service;

import com.pobitra.autocare.dto.BookingRequestDTO;
import com.pobitra.autocare.dto.BookingResponseDTO;
import com.pobitra.autocare.enums.BookingStatus;

import java.util.List;

public interface BookingService {

    BookingResponseDTO createBooking(
            BookingRequestDTO dto,
            String email
    );

    List<BookingResponseDTO> getAllBookings();

    BookingResponseDTO getBookingById(Long id);

    List<BookingResponseDTO> getMyBookings(
            String email
    );

    List<BookingResponseDTO> getBookingsByEmail(
            String email
    );

    List<BookingResponseDTO> getBookingsByStatus(
            BookingStatus status
    );

    BookingResponseDTO updateBookingStatus(
            Long id,
            BookingStatus status
    );

    void deleteBooking(
            Long id,
            String email
    );
}